### ISSUE TITLE
max_allowed_packet warning breaks Elasticsearch query 

### ISSUE THREAD

**Issue by ghost** [zammad/zammad#2186]
### Infos:

* Used Zammad version: 2.6.x
* Installation method (source, package, ..): packe 
* Operating system: Debian 9.5
* Database + version: mysqld 10.1.26-MariaDB-0+deb9u1
* Elasticsearch version: 5.6.10
* Browser + version: Independent from browser

### Expected behavior:

* Search output is displayed (Searchs Tickets, Companies etc.)

### Actual behavior:

* error is displayed: StatusCode: 500

### Steps to reproduce the behavior:

* Search

Yes I'm sure this is a bug and no feature request or a general question.

The max_allowed_packet-warning is added to every elasticsearch query, so the elasticsearch URL is broken. The max_allowed_packet is actually 128M, the maximum mailsize in Zammad 64MB. Fetching mails works fine.

Health says:

181 failing background jobs.
Failed to run background job #1 'BackgroundJobSearchIndex' 10 time(s) with 180 attempt(s).

+++

```
I, [2018-08-14T06:51:00.533252 #959]  INFO -- : Started GET "/api/v1/search?query=d&limit=10&_=1534221906304" for 5.100.32.106 at 2018-08-14 06:51:00 +0200
I, [2018-08-14T06:51:00.544432 #959]  INFO -- : Processing by SearchController#search_generic as JSON
I, [2018-08-14T06:51:00.544623 #959]  INFO -- :   Parameters: {"query"=>"d", "limit"=>"10", "_"=>"1534221906304"}
I, [2018-08-14T06:51:00.571756 #959]  INFO -- : # curl -X POST "Warning: Database config value 'max_allowed_packet' less than Zammad setting 'Maximum Email Size'
Zammad will fail to process emails (both incoming and outgoing)
larger than the value of 'max_allowed_packet' (64MB).
Please increase this value in your MariaDB configuration accordingly.
Warning: Database config value 'max_allowed_packet' less than Zammad setting 'Maximum Email Size'
Zammad will fail to process emails (both incoming and outgoing)
larger than the value of 'max_allowed_packet' (64MB).
Please increase this value in your MariaDB configuration accordingly.
http://127.0.0.1:9200/zammad_production/User/_search" \
E, [2018-08-14T06:51:00.572197 #959] ERROR -- : bad URI(is not URI?): Warning: Database config value 'max_allowed_packet' less than Zammad setting 'Maximum Email Size'
Zammad will fail to process emails (both incoming and outgoing)
larger than the value of 'max_allowed_packet' (64MB).
Please increase this value in your MariaDB configuration accordingly.
Warning: Database config value 'max_allowed_packet' less than Zammad setting 'Maximum Email Size'
Zammad will fail to process emails (both incoming and outgoing)
larger than the value of 'max_allowed_packet' (64MB).
Please increase this value in your MariaDB configuration accordingly.
http://127.0.0.1:9200/zammad_production/User/_search (URI::InvalidURIError)
/opt/zammad/vendor/ruby-2.4.4/lib/ruby/2.4.0/uri/rfc3986_parser.rb:67:in `split'
/opt/zammad/vendor/ruby-2.4.4/lib/ruby/2.4.0/uri/rfc3986_parser.rb:73:in `parse'
/opt/zammad/vendor/ruby-2.4.4/lib/ruby/2.4.0/uri/common.rb:231:in `parse'
/opt/zammad/lib/user_agent.rb:47:in `get'
/opt/zammad/lib/search_index_backend.rb:352:in `search_by_index'
/opt/zammad/lib/search_index_backend.rb:306:in `block in search'
/opt/zammad/lib/search_index_backend.rb:305:in `each'
/opt/zammad/lib/search_index_backend.rb:305:in `search'
/opt/zammad/app/controllers/search_controller.rb:65:in `search_generic'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_controller/metal/basic_implicit_render.rb:4:in `send_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/abstract_controller/base.rb:186:in `process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_controller/metal/rendering.rb:30:in `process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/abstract_controller/callbacks.rb:20:in `block in process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/callbacks.rb:131:in `run_callbacks'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/abstract_controller/callbacks.rb:19:in `process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_controller/metal/rescue.rb:20:in `process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/notifications.rb:166:in `block in instrument'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/notifications.rb:166:in `instrument'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_controller/metal/params_wrapper.rb:252:in `process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activerecord-5.1.5/lib/active_record/railties/controller_runtime.rb:22:in `process_action'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/abstract_controller/base.rb:124:in `process'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionview-5.1.5/lib/action_view/rendering.rb:30:in `process'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_controller/metal.rb:189:in `dispatch'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_controller/metal.rb:253:in `dispatch'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/routing/route_set.rb:49:in `dispatch'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/routing/route_set.rb:31:in `serve'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/journey/router.rb:50:in `block in serve'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/journey/router.rb:33:in `each'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/journey/router.rb:33:in `serve'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/routing/route_set.rb:844:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:190:in `call!'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/strategy.rb:168:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/omniauth-1.7.1/lib/omniauth/builder.rb:63:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/rack-2.0.5/lib/rack/etag.rb:25:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/rack-2.0.5/lib/rack/conditional_get.rb:25:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/rack-2.0.5/lib/rack/head.rb:12:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/rack-2.0.5/lib/rack/session/abstract/id.rb:232:in `context'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/rack-2.0.5/lib/rack/session/abstract/id.rb:226:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/cookies.rb:613:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/callbacks.rb:26:in `block in call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/callbacks.rb:97:in `run_callbacks'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/callbacks.rb:24:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/debug_exceptions.rb:59:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/railties-5.1.5/lib/rails/rack/logger.rb:36:in `call_app'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/railties-5.1.5/lib/rails/rack/logger.rb:24:in `block in call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/tagged_logging.rb:69:in `block in tagged'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/tagged_logging.rb:26:in `tagged'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/tagged_logging.rb:69:in `tagged'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/railties-5.1.5/lib/rails/rack/logger.rb:24:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/remote_ip.rb:79:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/request_id.rb:25:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/rack-2.0.5/lib/rack/method_override.rb:22:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/rack-2.0.5/lib/rack/runtime.rb:22:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/activesupport-5.1.5/lib/active_support/cache/strategy/local_cache_middleware.rb:27:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/executor.rb:12:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/actionpack-5.1.5/lib/action_dispatch/middleware/static.rb:125:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/rack-2.0.5/lib/rack/sendfile.rb:111:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/railties-5.1.5/lib/rails/engine.rb:522:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/puma-3.11.0/lib/puma/configuration.rb:225:in `call'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/puma-3.11.0/lib/puma/server.rb:624:in `handle_request'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/puma-3.11.0/lib/puma/server.rb:438:in `process_client'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/puma-3.11.0/lib/puma/server.rb:302:in `block in run'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/puma-3.11.0/lib/puma/thread_pool.rb:120:in `block in spawn_thread'
/opt/zammad/vendor/bundle/ruby/2.4.0/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
I, [2018-08-14T06:51:00.573089 #959]  INFO -- : Completed 500 Internal Server Error in 28ms (Views: 0.3ms | ActiveRecord: 0.8ms)
```

**Comment by MrGeneration** [comment]
Your log clearly says 

[...]
Please increase this value in your **MariaDB** configuration accordingly.
[...]

Please head over to our Community-Forum for technical Questions or help on setup (unless you're a paying customer, you then can use our E-Mail support). 

Please also include further information like "how Zammad has been setup", because "wrong url" for elasticsearch rather sounds like you missed something during configuration. Maybe our docs https://docs.zammad.org/en/latest/ can help you, too.


This tracker is being watched by many people that get updates on every comment that happens on this tracker.

**Comment by ghost** [comment]
As I wrote: "The max_allowed_packet is actually 128M, the maximum mailsize in Zammad 64MB", the setting is already bigger than 64MB in the mariadb config... And this setup has been working fine. Does Zammad cache the max_allowed_packet value somewhere?

**Comment by thorsteneckel** [comment]
@clicksports - please enter the [Zammad rails console](https://docs.zammad.org/en/latest/admin-console.html#start-zammad-s-rails-console) and execute the following command:
`ActiveRecord::Base.connection.execute('SELECT @@max_allowed_packet;').first.first`
and post the output here. Thanks!

**Comment by ghost** [comment]
Loading production environment (Rails 5.1.5)
irb(main):002:0* ActiveRecord::Base.connection.execute('SELECT @@max_allowed_packet;').first.first => 134217728

That's 128MB.

**Comment by thorsteneckel** [comment]
So I reformatted the log file from above and instead of making things clear it gets more confusing. The text:
```
Warning: Database config value 'max_allowed_packet' less than Zammad setting 'Maximum Email Size'
Zammad will fail to process emails (both incoming and outgoing)
larger than the value of 'max_allowed_packet' (64MB).
Please increase this value in your MariaDB configuration accordingly.
Warning: Database config value 'max_allowed_packet' less than Zammad setting 'Maximum Email Size'
Zammad will fail to process emails (both incoming and outgoing)
larger than the value of 'max_allowed_packet' (64MB).
Please increase this value in your MariaDB configuration accordingly.
```

is send to the Elasticsearch which is not able to handle the request and fails. The message gets printed when trying to start Zammad with a wrong DB configuration. You can see from the log file that this error comes from a request to the HTTP REST API of Zammad. However, I can't imagine a way other than user input for this text to get there.

Please reboot all Zammad services (application server, websocket server and scheduler) and check your logs again. If the issue still persists please open a topic over at the [community board](http://community.zammad.org/). Thanks!

**Comment by blue-hound** [comment]
I just had this happen to me as well. I read this thread and searched the database. In the settings table I found that the "Elasticsearch Endpoint URL" entry has current_state set to 

```
--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess
value: "{"error":"bad URI(is not URI?): \u001b[33mWarning: Database config value 'max_allowed_packet' less than Zammad setting 'Maximum Email Size'\n
Zammad will fail to process emails (both incoming and outgoing)\n
larger than the value of 'max_allowed_packet' (16MB).\n
Please increase this value in your MariaDB configuration accordingly.\n\u001b[0m/zammad_production/User/_search"}"
```

Once I set the value back to what was in the intial state column the error cleared and the background scheduler started running normally again. Confirmed with 

`zammad run rake searchindex:rebuild`

During my intial setup max_allowed_packets was set to 16MB and I increased my mail handling to 25MB. So somehow this error text is getting inserted into the database. Even after I increased max_packet_size I was still getting the error on the admin page but the new value wasn't reflected. 

Now that I've fixed the database entry, the problem has resolved.
